packages/client/src/getBalances.ts

```diff
@@ -1,31 +1,40 @@
 import ERC20_ABI from "@openzeppelin/contracts/build/contracts/ERC20.json";
-import { getEndpoints } from "@zetachain/networks/dist/src/getEndpoints";
 import { getAddress } from "@zetachain/protocol-contracts";
 import ZRC20 from "@zetachain/protocol-contracts/abi/zevm/ZRC20.sol/ZRC20.json";
 import { ethers } from "ethers";
 import { formatUnits } from "ethers/lib/utils";
 import fetch from "isomorphic-fetch";
 
-export const getForeignCoins = async () => {
-  const api = getEndpoints("cosmos-http", "zeta_testnet")[0]?.url;
-  const endpoint = `${api}/zeta-chain/fungible/foreign_coins`;
-  const response = await fetch(endpoint);
-  const data = await response.json();
-  return data.foreignCoins;
-};
+import { ZetaChainClient } from "./client";
 
-export const getSupportedChains = async () => {
-  const api = getEndpoints("cosmos-http", "zeta_testnet")[0]?.url;
-  const endpoint = `${api}/zeta-chain/observer/supportedChains`;
-  const response = await fetch(endpoint);
-  const data = await response.json();
-  return data.chains;
-};
+export interface TokenBalance {
+  balance: string;
+  chain_id: number | string;
+  chain_name: string;
+  coin_type: string;
+  contract?: string;
+  decimals: number;
+  id: string;
+  symbol: string;
+  ticker: string;
+  zrc20?: string;
+}
 
-export const getBalances = async (evmAddress: any, btcAddress = null) => {
+/**
+ * Get token balances of all tokens on all chains connected to ZetaChain.
+ *
+ * @param this - ZetaChainClient instance.
+ * @param options.evm EVM address
+ * @param options.btc Bitcoin address
+ * @returns
+ */
+export const getBalances = async function (
+  this: ZetaChainClient,
+  { evm, btc }: { btc?: string; evm: string }
```

Comment:
Ok, 1c75e0d

I was kind of hoping to keep param names less verbose as this is an object, so users would have to repeat them.

```ts
getBalances({evm: "0x4955a3F38ff86ae92A914445099caa8eA2B9bA32"})

// vs

getBalances({evmAddress: "0x4955a3F38ff86ae92A914445099caa8eA2B9bA32"})
```

Do you think using `address` type instead of string would suffice or do you think being verbose is better.

```ts
// For example:
export const getBalances = async function (
  this: ZetaChainClient,
  { evm, btc }: { evm: address, btc?: address }
)
```

Reply:
I would still add the "address" suffix even if it's more verbose, but totally optional.